### PR TITLE
fix(cli): Fix filters with json output for vuln host command

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Fix json output to should work in combination with any filtering flags

vuln host scan-pkg-manifest vuln_host.go:165 
vuln host list-cves vuln_host.go:207
vuln host show-assessment  vuln_host.go:342

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: cli/cmd/vuln_host.go [hunk 1]
```diff
@@ -211,67 +202,14 @@ with fixes:
 			}
 
 			response, err := cli.LwApi.Vulnerabilities.Host.ListCves()
+
```
Comment (afiune) [inline cli/cmd/vuln_host.go:205]: nit: better not have empty lines between the err and the `if` check.
```suggestion
```

File: cli/cmd/vuln_host.go [hunk 2]
```diff
@@ -595,18 +532,14 @@ func hostVulnPackagesTable(cves []api.HostVulnCVE, withHosts bool) ([][]string,
 	return out, ""
 }
 
-func hostVulnCVEsTable(cves []api.HostVulnCVE) ([][]string, string) {
-	var out [][]string
+func hostVulnCVEs(cves []api.HostVulnCVE) ([]api.HostVulnCVE, string) {
```
Comment (afiune) [inline cli/cmd/vuln_host.go:535]: if this function is filtering please add that to the name of the function just like you did for `filterHostVulnCVEs()`

File: cli/cmd/vuln_host.go [hunk 3]
```diff
@@ -616,60 +549,73 @@ func hostVulnCVEsTable(cves []api.HostVulnCVE) ([][]string, string) {
...
+				pkg.FixedVersion,
+				pkg.Namespace,
+				pkg.HostCount,
+				pkg.PackageStatus,
+				pkg.Status,
+			})
+		}
+	}
+
+	//order by total number of host
```
Comment (afiune) [inline cli/cmd/vuln_host.go:572]: nit: comments should always have space in between
```suggestion
	// order by the total number of host
```

File: cli/cmd/vuln_host.go [hunk 4]
```diff
@@ -949,13 +875,23 @@ func hostScanPackagesVulnDetailsTable(vulns []api.HostScanPackageVulnDetails) []
 			fixedVersion = vuln.FixInfo.FixedVersion
 		}
 
+		vuln.FixInfo.FixedVersion = fixedVersion
```
Comment (afiune) [inline cli/cmd/vuln_host.go:878]: ugh, I might need to understand why are we changing the API response here 🤔 , I don't think we should do that, could we sync up?

File: cli/cmd/vuln_host.go [hunk 5]
```diff
@@ -967,47 +903,39 @@ func hostScanPackagesVulnDetailsTable(vulns []api.HostScanPackageVulnDetails) []
...
-		}
-
-		added := false
-		for i := range out {
-			if out[i][1] == vuln.Severity &&
-				out[i][2] == vuln.OsPkgInfo.Pkg &&
-				out[i][3] == vuln.OsPkgInfo.PkgVer {
+func filterHostScanPackagesVulnPackages(vulns []api.HostScanPackageVulnDetails) filteredPackageTable {
+	var filteredPackages []packageTable
+	var aggregatedPackages []packageTable
```
Comment (afiune) [inline cli/cmd/vuln_host.go:908]: nit
```suggestion
	var (
		filteredPackages   []packageTable
		aggregatedPackages []packageTable
	)
```

File: cli/cmd/vuln_host.go [hunk 6]
```diff
@@ -967,47 +903,39 @@ func hostScanPackagesVulnDetailsTable(vulns []api.HostScanPackageVulnDetails) []
...
 
-		fixes := "0"
-		if vuln.HasFix() {
-			fixes = "1"
-		}
+		aggregatedPackages = aggregatePackagesWithHosts(aggregatedPackages, pack, false, vuln.HasFix())
+	}
+
+	totalUnfiltered := len(filteredPackages) + len(aggregatedPackages)
+	return filteredPackageTable{packages: aggregatedPackages, totalPackages: len(aggregatedPackages), totalUnfiltered: totalUnfiltered}
```
Comment (afiune) [inline cli/cmd/vuln_host.go:927]: nit: style and no need to instantiate `totalUnfiltered` variable.
```suggestion
	return filteredPackageTable{
		packages:        aggregatedPackages,
		totalPackages:   len(aggregatedPackages),
		totalUnfiltered: len(filteredPackages) + len(aggregatedPackages),
	}
```

File: cli/cmd/vuln_host.go [hunk 7]
```diff
@@ -1018,3 +946,119 @@ func hostScanPackagesVulnPackagesView(vulns []api.HostScanPackageVulnDetails) []
...
+		)
+	}
+	return nil
+}
+
+// Build the cli output for vuln host scan-package-manifest
+func buildVulnHostScanPkgManifestReports(response api.HostVulnScanPkgManifestResponse) error {
+	if len(response.Vulns) == 0 {
+		// @afiune add a helpful message, possible things are:
+		cli.OutputHuman("There are no vulnerabilities found.\n")
```
Comment (afiune) [inline cli/cmd/vuln_host.go:1049]: I wanted to change this for months, let us do it now that I remembered and we can:
```suggestion
		cli.OutputHuman(fmt.Sprintf("There are no vulnerabilities found! Time for %s\n", randomEmoji()))
```

**Review by afiune — CHANGES_REQUESTED** [review]

File: cli/cmd/vuln_host.go (hunk 4 above)
Comment (dmurray-lacework) [reply cli/cmd/vuln_host.go:877]: Good catch Salim, I think you're right. In fact I think 872-877 is redundant and can just be removed.

**Review by afiune — APPROVED** [review]
tenor-110317699

**Review by afiune — APPROVED** [review]
tenor-70719112

**Pull request merged**